src/components/SortButton.tsx

```diff
@@ -0,0 +1,88 @@
+import { useEffect, useRef, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {
+  faArrowDownShortWide,
+  faSortDown,
+  faSortUp
+} from '@fortawesome/free-solid-svg-icons';
+
+import Button from './Button';
+import { todosActions } from '../store/store';
+
+import classes from './SortButton.module.css';
+
+function SortButton() {
+  const dispatch = useDispatch();
+  const [open, setOpen] = useState(false);
+  const ref = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    let handler = (e: any) => {
+      if (!ref.current?.contains(e.target as Node)) {
+        setOpen(false);
+      }
+    };
+    document.addEventListener('mousedown', handler);
+    return () => {
+      document.removeEventListener('mousedown', handler);
+    };
+  });
+
+  const sortByDateHandler = (order: string) => {
+    dispatch(todosActions.sortBy({ type: 'date', order: order }));
+    setOpen(false);
+  };
+  const sortByTextHandler = (order: string) => {
+    dispatch(todosActions.sortBy({ type: 'text', order: order }));
+    setOpen(false);
+  };
+
+  const { sort, dropdown, dropdownHeader, dropdownList, dropdownItem, sortUp } =
+    classes;
+
+  return (
+    <div ref={ref}>
+      <Button className={sort} onClick={() => setOpen(!open)}>

```

Comment:
I prefer to make separate functions, to not use callbacks inside onCkick, onChange e.t.c. The reason is if you will need to add some more logic to this handler, it will be much easier to do. Now You have only sortByText... but if you will need to add smth  else? there will be a lot of logic inside jsx.